contrib/charts/dragonfly/templates/_pod.tpl

```diff
@@ -64,39 +83,17 @@ containers:
     {{- end }}
     {{- with .Values.resources }}
     resources:
-      {{- toYaml . | nindent 6 }}
+      {{- toYaml . | trim | nindent 6 }}
     {{- end }}
-    {{- include "dragonfly.volumemounts" . | nindent 4 }}
+    {{- include "dragonfly.volumemounts" . | trim | nindent 4 }}
+    {{- if .Values.passwordFromSecret.enable -}}
     env:
-    {{- if .Values.passwordFromSecret.enable }}
       - name: DFLY_PASSWORD
         valueFrom:
           secretKeyRef:
             name: {{ .Values.passwordFromSecret.existingSecret.name }}
             key: {{ .Values.passwordFromSecret.existingSecret.key }}
     {{- end }}
-  {{- with .Values.extraContainers }}
```

Comment:
To prevent extra empty lines being rendered, This was moved above along while adding `trim` before we render.